compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -1074,11 +1075,41 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         //     fn f() {}
                         //   - f(0, 1,)
                         //   + f()
-                        if only_extras_so_far
-                            && !errors
-                                .peek()
-                                .is_some_and(|next_error| matches!(next_error, Error::Extra(_)))
-                        {
+                        let trim_next_comma = {
+                            if only_extras_so_far {
+                                if let Some(next_error) = errors.peek() {
+                                    match next_error {
```

Comment:
This can be squashed into a single `match` on `errors.peek()`.

(Technically even the `only_extras_so_far` check can be moved into that match as `_ if !only_extras_so_far => false`.)

Reply:
Thank you ideas, this inspire me to gather them to one `match` pattern